### PR TITLE
wavetool.py: add THD+N analysis for audio quality check

### PR DESCRIPTION
**Pull request by aiChaoSONG**
This PR add a generic THD+N analysis for audio quality check, and a specific alsabat audio check. for alsabat audio check, the ramp data at the beginning will be skipped.

### CODE REVIEW

File: tools/wavetool.py [hunk 1]
```diff
@@ -242,11 +288,11 @@ def parse_cmdline():
     help='sample bits of generated wave')
     parser.add_argument('-o', '--output', type=str, help='path to store generated files', default='.')
     # wave comparison arguments
-    parser.add_argument('-a', '--analyze', type=str, choices=['smart_amp', 'wov'], help='analyze recorded wave to give case verdict')
+    parser.add_argument('-a', '--analyze', type=str, choices=['smart_amp', 'wov', 'thdn', 'alsabat'], help='analyze recorded wave to give case verdict')
     parser.add_argument('-R', '--recorded_wave', type=str, help='path of recorded wave')
     parser.add_argument('-Z', '--zero_threshold', type=float, default=-50.3, help='zero threshold in dBFS')
     parser.add_argument('-H', '--hb_time', type=float, default=2.1, help='history buffer size')
-    parser.add_argument('-T', '--threshold', type=float, default=-72.0, help='expected threshold')
+    parser.add_argument('-T', '--threshold', type=float, default=-65.0, help='expected threshold')
```
Comment (xiulipan) [inline tools/wavetool.py:295]: Any comment about this default value change?

File: tools/wavetool.py [hunk 2]
```diff
@@ -205,6 +212,29 @@ def analyze_wav_wov(wave, fs):
         sys.exit(1002)
     print('wave analysis result: PASSED')
 
+def analyze_wav_alsabat(wave, fs_wav):
+    # Drop the ramp up data in unstable state
+    time_skip = 0.25
```
Comment (xiulipan) [inline tools/wavetool.py:217]: What is this value used here? Global value?

File: tools/wavetool.py [hunk 3]
```diff
@@ -242,11 +288,11 @@ def parse_cmdline():
     help='sample bits of generated wave')
     parser.add_argument('-o', '--output', type=str, help='path to store generated files', default='.')
     # wave comparison arguments
-    parser.add_argument('-a', '--analyze', type=str, choices=['smart_amp', 'wov'], help='analyze recorded wave to give case verdict')
+    parser.add_argument('-a', '--analyze', type=str, choices=['smart_amp', 'wov', 'thdn', 'alsabat'], help='analyze recorded wave to give case verdict')
```
Comment (xiulipan) [inline tools/wavetool.py:291]: what is alsabat mode here? Does it means we have some predefined parameters here

**Review by xiulipan — CHANGES_REQUESTED** [review]
I think we can make this feature more generic. Have some parameter for the skip things and add some predefined configure for some existing test.

File: tools/wavetool.py (hunk 1 above)
Comment (aiChaoSONG) [reply tools/wavetool.py:295]: The first version I made it -72dB, but this threshold is high for our use case. eg, wov, so I have to lower the default value.

File: tools/wavetool.py (hunk 2 above)
Comment (aiChaoSONG) [reply tools/wavetool.py:217]: There is a ramp in recorded wave in alsabat test case, and the duration is about 0.25 second

File: tools/wavetool.py (hunk 3 above)
Comment (aiChaoSONG) [reply tools/wavetool.py:291]: No, The alsabat mode here means that we will skip the ramp, and start analysis from stable state.

Comment (xiulipan) [reply tools/wavetool.py:291]: OK, so this is only some configuration for test right? It is not some algorithm like `thdn`, right? So can we make the configurations more straightforward in some static format data.

**Comment by aiChaoSONG** [comment]
close first, for no time to pick up this PR, will reopen later

**Pull request closed without merging**